crates/router/src/core/customers.rs

```diff
@@ -176,17 +243,53 @@ pub async fn delete_customer(
         }?,
     };
 
+    let key = get_key_and_algo(&**db, merchant_account.merchant_id.clone())
+        .await
+        .change_context(errors::ApiErrorResponse::InternalServerError)?;
+
     let update_address = storage::AddressUpdate::Update {
         city: Some(REDACTED.to_string()),
         country: None,
-        line1: Some(REDACTED.to_string().into()),
-        line2: Some(REDACTED.to_string().into()),
-        line3: Some(REDACTED.to_string().into()),
-        state: Some(REDACTED.to_string().into()),
-        zip: Some(REDACTED.to_string().into()),
-        first_name: Some(REDACTED.to_string().into()),
-        last_name: Some(REDACTED.to_string().into()),
-        phone_number: Some(REDACTED.to_string().into()),
+        line1: Some(
+            Encryptable::encrypt(REDACTED.to_string().into(), &key, GcmAes256 {})
+                .await
+                .change_context(errors::ApiErrorResponse::InternalServerError)?,
```

Comment:
We don't want to handle InternalServerError like this.

you can create a closure which returns Result<X,Y>, Y can be discarded with underscore and just do a change_context once.

```
let update_addressupdate_address = || -> Result<_, _> {
	...
	line1: Some( Encryptable::encrypt(REDACTED.to_string().into(), &key, GcmAes256 {})
                .await?;
    ...
}().change_context(errors::ApiErrorResponse::InternalServerError)
```

Reply:
Other places also do this change